test/unit/connection-pool.test.js

```diff
@@ -74,6 +74,14 @@ test('API', t => {
     }, 10)
   })
 
+  t.test('markDead should ignore connections that no longer exists', t => {
+    const pool = new ConnectionPool({ Connection, sniffEnabled: true })
+    pool.addConnection('http://localhost:9200/')
+    pool.markDead({ id: 'foo-bar' })
+    t.deepEqual(pool.dead, [])
+    t.end()
+  })
+
```

Comment:
Should be the same tested for `markAlive`?
Is it ok to `markAlive` a connection that isn't in the list of `connections`?

Reply:
There's no need to add it there, as both `markAlive` and `markDead` are just modifying the given connection instance.
The only place where `markDead` is actually doing something "dangerous", is when updating the `dead` array, which is just a cache of the dead nodes.

I'm checking the existence of a connection only when modifying the `dead` array to skip useless loops :)